### PR TITLE
Add delete to saved pins

### PR DESCRIPTION
**Pull request by steveone**
## Issue Description:
Add delete functionality to frontend savedpins

### Summary of solution:
1.   Added option textbook to create a description when saving new pins (needs css work)
2. Added display of description and delete button when saved pins showing on map are clicked on
3. Added code to have the server delete the saved marker from the database and remove it from the map

### Can this issue be closed?

Additional CSS work to prettify is needed but that can be in another issue if desired

### Should any new issues be added as a result of this solution?

CSS work needed to prettify the input box on new markers and the description and delete button on the saved marker popups

### Thanks for contributing!

### CODE REVIEW

**Comment by steveone** [comment]
No actual changes were made to the following files:

-  controllers/utils-controller.js
-  controllers/saved-pins-controller.js
-  client/src/js/server-requests-utils.js

The changes shown are whitespace caused by debugging, I'm not sure how to remove those changes from being seen as changes by github.

Additional issue:

There is now two duplicated (createLabel and createButton) functions in 

-  client/src/components/LeafletOsmMap/index.js
-  client/src/js/saved-places.js

that can probably be moved to a utility file to avoid duplicate code.

**Comment by steveone** [comment]
@jkwening I hijacked "place_id" based on a "slack" conversation with @motosharpley. The idea is to automatically populate the places data as a future feature so maybe a rename to "Desc" should be held off or perhaps we should add a new "Desc" option to supplement the eventual "place_id"? I'm open to either.

**Comment by jkwening** [comment]
Adding a new "desc" field is fine with me instead of replacing "place_id" but we definitely shouldn't mislabel the data we're storing. That's bound to cause confusion later on, especially for others who aren't as familiar with the code base.

**Comment by steveone** [comment]
@jkwening, I will try to work on that tomorrow if not in the next few days. Thanks for the pointers to what areas will need updating

**Comment by steveone** [comment]
@jkwening I've updated the code to use a "desc" field instead of "place_id" for the user descriptions. The mongoose model and route code have been updated as well. It tested well on my side. Let me know if you have any other thoughts (I know I still need to fix the CSS)

File: client/src/js/saved-places.js [hunk 1]
```diff
@@ -61,20 +61,23 @@ const makePinMarkers = (pinArray = [], L, markerOptions=FAV_MARKER_OPTIONS) => {
...
         riseOnHover: true,
-        title: pin.place_id
+        title: pin.place_id,
+        desc: pin.desc
       });
     }
-    
-    thisMarker.data = pin; //save the db data to the marker
 
+    const container = L.DomUtil.create('div');
```
Comment (jkwening) [inline client/src/js/saved-places.js]: `container` is unused within the scope of `makePinMarkers()` - it looks like it should be deleted? It looks like it actually implemented at line 112.

Comment (steveone) [reply client/src/js/saved-places.js]: You would be correct. I think I originally had logic there and had to move it down. Great catch.

**Pull request merged**